irr/include/irrString.h

```diff
@@ -166,21 +167,29 @@ class string
 
 	//! Assignment operator for strings, ASCII and Unicode
 	template <class B>
-	string<T> &operator=(const B *const c)
+	string<T> &operator=(const B * c)
 	{
 		if (!c) {
 			clear();
 			return *this;
 		}
-
-		// no longer allowed!
-		_IRR_DEBUG_BREAK_IF((void *)c == (void *)c_str());
+		
+		if constexpr (sizeof(T) != sizeof(B)) {
```

Comment:
Now that we have this check we can also put the `if ((void *)c == (void *)c_str()) return *this;` optimization back in

Reply:
It will be added back my son.

Reply:
Thanks father